bluemira/radiation_transport/flux_surfaces_maker.py

```diff
@@ -0,0 +1,303 @@
+            self.eq.x, self.eq.z, self.eq.psi(), x, z, self.eq.psi(x, z)
+        )
+        coords = Coordinates({"x": coords[0], "z": coords[1]})
+        f_s = OpenFluxSurface(coords)
+        lfs, hfs = f_s.split(self._o_point, plane=self._yz_plane)
+        return lfs, hfs
+
+    def _make_flux_surfaces_ob(self):
+        """
+        Make the flux surfaces on the outboard.
+        """
+        self.x_sep_omp, x_out_omp = self._get_sep_out_intersection(outboard=True)
+
+        self.flux_surfaces_ob_lfs = []
+        self.flux_surfaces_ob_hfs = []
+
+        for x in np.arange(self.x_sep_omp + self.dx_mp, x_out_omp - EPS, self.dx_mp):
+            lfs, hfs = self._make_flux_surfaces(x, self._o_point.z)
+            self.flux_surfaces_ob_lfs.append(lfs)
+            self.flux_surfaces_ob_hfs.append(hfs)
+
+    def _make_flux_surfaces_ib(self):
+        """
+        Make the flux surfaces on the inboard.
+        """
+        self.x_sep_imp, x_out_imp = self._get_sep_out_intersection(outboard=False)
+
+        self.flux_surfaces_ib_lfs = []
+        self.flux_surfaces_ib_hfs = []
+
+        for x in np.arange(self.x_sep_imp - self.dx_mp, x_out_imp + EPS, -self.dx_mp):
+            lfs, hfs = self._make_flux_surfaces(x, self._o_point.z)
+            self.flux_surfaces_ib_lfs.append(lfs)
+            self.flux_surfaces_ib_hfs.append(hfs)    
+
+    def _clip_flux_surfaces(self, first_wall):
+        """
+        Clip the flux surfaces to a first wall. Catch the cases where no intersections
+        are found.
+        """
+        for group in [
+            self.flux_surfaces_ob_lfs,
+            self.flux_surfaces_ob_hfs,
+            self.flux_surfaces_ib_lfs,
+            self.flux_surfaces_ib_hfs,
+        ]:
+            if group:
+                for i, flux_surface in enumerate(group):
+                    flux_surface.clip(first_wall)
+                    if flux_surface.alpha is None:
+                        # No intersection detected between flux surface and first wall
+                        # Drop the flux surface from the group
+                        group.pop(i)
+
+    def analyse(self, first_wall):
+        """
+        Perform the calculation to obtain charged particle heat fluxes on the
+        the specified first_wall
+
+        Parameters
+        ----------
+        first_wall: Coordinates
+            The closed first wall geometry on which to calculate the heat flux
+
+        Returns
+        -------
+        x: np.array
+            The x coordinates of the flux surface intersections
+        z: np.array
+            The z coordinates of the flux surface intersections
+        heat_flux: np.array
+            The perpendicular heat fluxes at the intersection points [MW/m^2]
+        """
+        self.first_wall = self._process_first_wall(first_wall)
+
+        if self.eq.is_double_null:
+            self._analyse_DN(first_wall)
+        else:
+            self._analyse_SN(first_wall)
```

Comment:
dx_omp should be returned from the single null and both dx_omp and dx_impo should be returned from the double null not stored on the class
```suggestion
        if self.eq.is_double_null:
            return self._analyse_DN(first_wall)
        else:
            return self._analyse_SN(first_wall)
```